### PR TITLE
Improves detection of pencil on all platforms

### LINKED ISSUE #123: Detect whether iOS device has pencil or not
We need a way to detect whether an apple device has a pencil or not. This is not relevant to android because the logic already exists in the original artmaker

### PR DESCRIPTION
**Pull request by Fbada006**
closes #123 

This adds the ability to detect the presence of an s-pen in apple, although it is untested on iOS. However, the logic has been changed to be handled the same as in android, which has been tested and works as expected. 🎉

With this new logic, we will assume that no device has a stylus. Once a user draws at least once with a stylus, we will toggle an internal pref that there is a stylus available, at which point two things will happen:
1. They get a dialog indicating that there is stylus support in the lib and that it offers a better drawing experience
2. The stylus settings will now become visible

### CODE REVIEW

File: androidApp/src/main/java/com/artmaker/androidApp/MainActivity.kt [hunk 1]
```diff
@@ -34,7 +33,6 @@ class MainActivity : ComponentActivity() {
                 ArtMaker(
                     modifier = Modifier.fillMaxSize()
                 )
-                println("Greetings----${Greeting().greet()}")
```
Comment (Fbada006) [inline androidApp/src/main/java/com/artmaker/androidApp/MainActivity.kt:37]: Not necessary

File: shared/src/commonMain/kotlin/com/fbada006/shared/ArtMakerViewModel.kt [hunk 2]
```diff
@@ -149,33 +151,9 @@ internal class ArtMakerViewModel(
         _backgroundImage.value = bitmap
     }
 
-    private fun selectStrokeWidth(strokeWidth: Int) {
```
Comment (Fbada006) [inline shared/src/commonMain/kotlin/com/fbada006/shared/ArtMakerViewModel.kt:152]: Did some refactoring here

File: shared/src/commonMain/kotlin/com/fbada006/shared/composables/ArtMakerDrawScreen.kt [hunk 3]
```diff
@@ -162,12 +162,15 @@ internal fun ArtMakerDrawScreen(
 
                         when (event.type) {
                             PointerEventType.Press -> {
-                                getDialogType(change, state.shouldUseStylusOnly)?.let { type ->
+                                // Only update the stylus availability if this is a stylus input and the stylus availability state is not updated
+                                if (change.isStylusInput() && !state.isStylusAvailable) onAction(ArtMakerAction.UpdateStylusAvailability(change.isStylusInput()))
```
Comment (Fbada006) [inline shared/src/commonMain/kotlin/com/fbada006/shared/composables/ArtMakerDrawScreen.kt:166]: This is where the detection actually occurs. This should handle both on iOS and android the same now

File: shared/src/commonMain/kotlin/com/fbada006/shared/models/ArtMakerConfiguration.kt [hunk 4]
```diff
@@ -36,7 +36,7 @@ data class ArtMakerConfiguration(
     val strokeSliderTextColor: Color = Color.Unspecified,
     val pickerCustomColors: List<Color> = emptyList(),
     val canvasBackgroundColor: Int = Color.White.toArgb(),
-    val controllerBackgroundColor: Color = Color.Unspecified,
+    val controllerBackgroundColor: Color = Color.LightGray.copy(alpha = 0.5f),
```
Comment (Fbada006) [inline shared/src/commonMain/kotlin/com/fbada006/shared/models/ArtMakerConfiguration.kt:39]: This should remove the line effect we had on ios @CalebKL

File: shared/src/commonMain/kotlin/com/fbada006/shared/models/ArtMakerConfiguration.kt [hunk 5]
```diff
@@ -36,7 +36,7 @@ data class ArtMakerConfiguration(
     val strokeSliderTextColor: Color = Color.Unspecified,
     val pickerCustomColors: List<Color> = emptyList(),
     val canvasBackgroundColor: Int = Color.White.toArgb(),
-    val controllerBackgroundColor: Color = Color.Unspecified,
+    val controllerBackgroundColor: Color = Color.LightGray,
```
Comment (Fbada006) [inline shared/src/commonMain/kotlin/com/fbada006/shared/models/ArtMakerConfiguration.kt:39]: This should remove the line effect we had on ios @CalebKL

File: shared/src/commonMain/kotlin/com/fbada006/shared/composables/ArtMakerDrawScreen.kt (hunk 3 above)
Comment (CalebKL) [reply shared/src/commonMain/kotlin/com/fbada006/shared/composables/ArtMakerDrawScreen.kt:166]: Okay. This makes sense

**Review by CalebKL — APPROVED** [review]

**Review by CalebKL — APPROVED** [review]

**Review by CalebKL — APPROVED** [review]

**Review by CalebKL — APPROVED** [review]
Okay

**Pull request merged**